### ISSUE TITLE
Error when building with extensions

### ISSUE THREAD

**Issue by schlista** [defold/defold#9717]
### Expected behaviour
Program to build without error

### Actual behaviour
When using defos after I build once I get an error and I have to delete the build directory to get it to build again.  In testing I found if I add an app manifest I always get the error.

### Steps to reproduce

Easiest way to reproduce it is to add an app manifest and then hit F5.
Then you'll get.
![image](https://github.com/user-attachments/assets/b41dd7d2-0d42-4497-89a0-0e74ca9dc1ad)

If you don't get it the first time try building a 2nd or 3rd time.

I was able to repro the issue with defos OS example as well.  It would work the first time and then usually give the error upon 2nd build.  Sometimes it would take a couple tries.

<hr/>

<table>
<tr><td>Build time</td><td>2024-11-08T09:48:06.519625</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>415b7a13d96c14dcce3239089b33736cfba4325e</td></tr>
<tr><td>Defold engine sha</td><td>edfdbe31830c1f8aa4d96644569ae87a8ea32672</td></tr>
<tr><td>Defold version</td><td>1.9.4</td></tr>
<tr><td>GPU</td><td>NVIDIA GeForce RTX 3080 Ti/PCIe/SSE2</td></tr>
<tr><td>GPU Driver</td><td>4.6.0 NVIDIA 566.03</td></tr>
<tr><td>Java version</td><td>17.0.5+8</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Windows 11</td></tr>
<tr><td>OS version</td><td>10.0</td></tr>
<table>

**Comment by AGulev** [comment]
Are you sure you close executable after the first run? 
Did you check task manager?

**Comment by schlista** [comment]
Yes.  Also I even closed defold and it still occurs when I opened it. Also removing the manifest let's it build without issue. 
-------- Original message --------From: Alexey Gulev ***@***.***> Date: 11/10/24  12:49 PM  (GMT-05:00) To: defold/defold ***@***.***> Cc: schlista ***@***.***>, Author ***@***.***> Subject: Re: [defold/defold] Error when building with extensions (Issue #9717) 
Are you sure you close executable after the first run?
Did you check task manager?

**Comment by AGulev** [comment]
I'm talking not about the editor, but dmengine. Are you sure you don't have opened copy that hangs and stay opend in memory?
![CleanShot 2024-11-10 at 21 26 51@2x](https://github.com/user-attachments/assets/4337be15-fe2a-4444-82ec-e68137b0ea0c)

Did you try to restart your computer.

**Comment by schlista** [comment]
Correct.  it's not running.  In fact, I rebooted my machine and launched defold.  Tried to run the defos example and it failed with the same error.  I have to delete the build folder to get it to run.
________________________________

**Comment by AGulev** [comment]
Is it still an issue for you?

**Comment by schlista** [comment]
I just tested the sample and it worked without issue.  Thank you.
